### PR TITLE
[V6] WalletAccount

### PR DESCRIPTION
**Pull request by tabaktoni**
## Motivation and Resolution

https://community.starknet.io/t/snip-standardizing-starknet-wallet-connections-with-write-api-and-json-rpc-integration/98049#using-rpc-for-starknet-wallet-connections-1-6

- [x] invoke
- [x] declare
- [x] deploy
- [x] events

Declare had some error on extension side to discuss.

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [ ] Performed a self-review of the code
- [ ] Rebased to the last commit of the target branch (or merged it into my branch)
- [ ] Linked the issues which this PR resolves
- [ ] Documented the changes in code (API docs will be generated automatically)
- [ ] Updated the tests
- [ ] All tests are passing

### CODE REVIEW

**Comment by tabaktoni** [comment]
> @tabaktoni Just a feedback on the draft WalletAccount PR. Even though having class based approach for other starknet.js entities makes sense, for WalletAccount in particular I think we should go with functional approach. WalletAccount or WalletClient should just be an “interface” providing an high level interfacing between get-starknet and wallets, I like viem’s approach to this.

The initial idea is a drop-in replacement for Account Class. But we can eventually do both

File: src/wallet/account.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    const declareContractPayload = extractContractHashes(payload);
+
+    // DISCUSS: HOTFIX: Adapt Abi format
+    const pContract = payload.contract as CompiledSierra;
+    const cairo1Contract = {
+      ...pContract,
+      abi: stringify(pContract.abi),
+    };
+
+    // Check FIx
```
Comment (ivpavici) [inline src/wallet/account.ts:146]: is this checked or still a TODO?

File: src/wallet/account.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    };
+
+    return addDeployAccountTransaction(this.walletProvider, params);
+  }
+
+  override signMessage(typedData: TypedData) {
+    return signMessage(this.walletProvider, typedData);
+  }
+
+  // MISSING ESTIMATES
```
Comment (ivpavici) [inline src/wallet/account.ts:187]: maybe add a TODO: for easier searching?

File: src/wallet/connect.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  AddStarknetChainParameters,
+  NetworkChangeEventHandler,
+  StarknetChainId,
+  StarknetWindowObject,
+  TypedData,
+  WatchAssetParameters,
+} from './getst/main';
+
+/**
+ * Request Permission for wallet account, return addresses that's allowed by user
```
Comment (ivpavici) [inline src/wallet/connect.ts:15]:
```suggestion
 * Request Permission for wallet account, return addresses that are allowed by user
```

File: src/wallet/connect.ts [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  });
+}
+
+/**
+ * Request adding custom Starknet chain
+ * @param chain AddStarknetChainParameters
+ * @returns boolean
+ */
+export function addStarknetChain(swo: StarknetWindowObject, chain: AddStarknetChainParameters) {
+  // Can this set custom RPC endpoint ?
```
Comment (ivpavici) [inline src/wallet/connect.ts:54]: also maybe TODO: here if still relevant

File: src/wallet/getst/localStorageStore.d.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export interface IStorageWrapper {
+  set(value: string | null | undefined): boolean;
+  get(): string | null | undefined;
+  delete(): boolean;
+}
+export declare class LocalStorageWrapper implements IStorageWrapper {
+  #private;
```
Comment (ivpavici) [inline src/wallet/getst/localStorageStore.d.ts:7]: is this a leftover?

File: src/wallet/types.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { RpcMessage, StarknetWindowObject } from './getst/main';
+
+// ---- TT Request Handler
+export type RpcCall = Omit<RpcMessage, 'result'>;
+
+// -- TT Better naming
```
Comment (ivpavici) [inline src/wallet/types.ts:6]: maybe a TODO: here ?

File: src/wallet/account.ts (hunk 1 above)
Comment (tabaktoni) [reply src/wallet/account.ts:146]: Yes it is diff between declare and api-wallet declare, the one Dhruv mentioned on call before Denver.
I would leave it for now as reminder

File: src/wallet/connect.ts (hunk 4 above)
Comment (tabaktoni) [reply src/wallet/connect.ts:54]: Yeah this one didn't work on the custom wallet build

File: src/wallet/getst/localStorageStore.d.ts (hunk 5 above)
Comment (tabaktoni) [reply src/wallet/getst/localStorageStore.d.ts:7]: It is a prop from get-starknet but now mentioning it, it looks like I need to update that one.

**Review by ivpavici — APPROVED** [review]

**Pull request merged**